source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,378 @@
+                                if (cell->level() >= isotherms_i[isotherm_line][1])
+                                  {
+                                    clear_refine = true;
+                                  }
+                                // If the current refinement level is larger then the maximum level, a coarsening flag should be placed, and we don't have to look any further.
+                                if (cell->level() >  isotherms_i[isotherm_line][1])
+                                  {
+                                    coarsen = true;
+                                    break;
+                                  }
+
+                              }
+                          }
+                      }
+                  }
+
+                /// Perform the actual placement of the coarsening and refinement flags
+                if (clear_refine)
+                  cell->clear_refine_flag ();
+                if (coarsen)
+                  cell->set_coarsen_flag ();
+                if (clear_coarsen)
+                  cell->clear_coarsen_flag ();
+                if (refine)
+                  cell->set_refine_flag ();
+              }
+        }
+    }
+
+    template <int dim>
+    void
+    Isotherms<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+
+        prm.enter_subsection("Isotherms");
+        {
+
+          prm.declare_entry ("Exclude composition", "-1",
+                             Patterns::Double (),
+                             "Number of mininum and maximum areas defined by isotherms are going to be defined."
+                            );
+          prm.declare_entry ("List of isotherms", "",
+                             Patterns::List(Patterns::List(Patterns::Anything(),0,100000000,","),0,100000000,";"),
+                             "Number of mininum and maximum areas defined by isotherms are going to be defined."
+                            );
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Isotherms<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+        minimum_refinement_level = Utilities::string_to_int(prm.get ("Minimum refinement level"));
+        maximum_refinement_level = Utilities::string_to_int(prm.get("Initial global refinement")) + Utilities::string_to_int(prm.get("Initial adaptive refinement"));
+        prm.enter_subsection("Isotherms");
+        {
+          exclude_composition = Utilities::string_to_int(prm.get("Exclude composition"));
+          std::vector<std::string> isotherms_outer_loop = Utilities::split_string_list(prm.get ("List of isotherms"),';');
+          // process the List of isotherms to get all data out of the structure
+          isotherms.resize(isotherms_outer_loop.size(),std::vector<std::string>(4,"0"));
```

Comment:
All of the conversions from string to double I mentioned above should already be happening in this function.